src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -81,11 +106,132 @@ public void Dispose()
     {
         _secretMasker?.Dispose();
         _secretMasker = null;
+        _telemetry = null;
     }
 
     public string MaskSecrets(string input)
     {
-        return _secretMasker.MaskSecrets(input);
+        _telemetry?.ProcessInput(input);
+        return _secretMasker.MaskSecrets(input, _telemetry?.DetectionAction);
+    }
+
+    public void EnableTelemetry()
+    {
+        _telemetry ??= new Telemetry();
+    }
+
+    /// <summary>
+    /// If enabled via <see cref="TelemetryEnabled"/> opt-in, publishes
+    /// telemetry via the provided callback that handles sending it over the
+    /// wire.
+    ///
+    /// Always publishes a 'SecretMasker' event with overall stats, and may
```

Comment:
>publishes [](http://example.com/codeflow?start=15&length=9)

this is getting to a suggestive level of detail without committing fully to it. :) I think I might be tempted to build out the concise summary a bit more, then add a <remarks> node that describes every data point in the overall stats that are published, viz. overall time of scan, count of raw detections, unique number of exposed secrets (by c3id), # of string input processed, and overall length of content scanned by char count.

Reply:
I've removed this comment, but probably something should come back and take this feedback into account.

Reply:
I've decided to move details into implementation comments. Repeating the whole schema in docs is likely to be a maintenance burden.